editor/plugins/editor_preview_plugins.cpp

```diff
@@ -851,7 +851,11 @@ Ref<Texture2D> EditorFontPreviewPlugin::generate_from_path(const String &p_path,
 
 	Ref<Font> font = sampled_font;
 
-	font->draw_string(canvas_item, pos, sample, HORIZONTAL_ALIGNMENT_LEFT, -1.f, 50, Color(1, 1, 1));
+	const Color c = GLOBAL_GET("rendering/environment/defaults/default_clear_color");
+	const float luminance = 0.2126 * c.r + 0.7152 * c.g + 0.0722 * c.b;
```

Comment:
I'm surprised that we wouldn't have something like this in `Color` already.

Does that mean that other parts of the adaptive theme coloring also hardcode similar checks, or use a different metric? CC @pycbouh @Calinou

Reply:
Indeed, a different approach is used, the color components are averaged and that value is checked:

```cpp
bool EditorSettings::is_dark_theme() {
	int AUTO_COLOR = 0;
	int LIGHT_COLOR = 2;
	Color base_color = get("interface/theme/base_color");
	int icon_font_color_setting = get("interface/theme/icon_and_font_color");
	return (icon_font_color_setting == AUTO_COLOR && ((base_color.r + base_color.g + base_color.b) / 3.0) < 0.5) || icon_font_color_setting == LIGHT_COLOR;
}
```

I'm not strong with the color theory, but it makes sense to unify this somehow. What would the `Color` method be? `is_dark()/is_light()` that returns true depending on the luminance factor compared to `0.5`?

Reply:
Some people use rgb, some use luminance, personally I just hardcoded things, but this is definitely something worth unifying and having built-in (AND exposed to scripting!)